packages/ast/api.d.ts

```diff
@@ -44,11 +44,18 @@ declare interface XMLProlog {
   readonly position: SourcePosition;
 }
 
+// A Prefix cannot include a colon ":" so the below string will never conflict
+// with a valid namespace prefix.
+declare type DefaultNS = "::DEFAULT";
+declare const DefaultNS: DefaultNS;
+declare type Prefix = string | DefaultNS;
+declare type Uri = string;
+
 declare interface XMLElement {
   readonly type: "XMLElement";
   readonly parent: XMLElement | XMLDocument;
 
-  readonly namespaces: { prefix?: string; uri: string }[];
+  readonly namespaces: Record<Prefix, Uri>;

```

Comment:
Should change this to a Real ECMAScript Map
as a plain object still has some properties and methods (e.g `toString`) via inheritance and users will not perform `hasOwnProperty` before accessing the "pesudo" map

Reply:
Can use `Object.create(null)` as well